src/accelerate/test_utils/scripts/test_script.py

```diff
@@ -400,6 +400,33 @@ def check_seedable_sampler_in_batch_sampler_shard():
     ), "Sampler in BatchSamplerShard is not SeedableRandomSampler."
 
 
+def check_seedable_sampler_with_data_seed():
```

Comment:
I'm not sure if I quite get this test. I ran it with the current main branch, i.e. without the fix, and it passed. Of course, I removed the `data_seed=data_seed` argument, as it doesn't exist. But should the test not fail on main?

Reply:
That's the weird part. Not sure what writing a test for this would look like tbh since it's just a passthrough in-a-sense. I guess check that the output when using it vs not differ?

Reply:
Better test added 🫡